sequencer/src/persistence/fs.rs

```diff
@@ -222,41 +222,36 @@ impl Inner {
     fn collect_garbage(
         &mut self,
         view: ViewNumber,
-        intervals: &[RangeInclusive<u64>],
+        intervals: &[RangeInclusive<ViewNumber>],
     ) -> anyhow::Result<()> {
-        let view_number = view.u64();
-        let prune_view = view.saturating_sub(self.view_retention);
-
-        let delete_files =
-            |intervals: &[RangeInclusive<u64>], keep, dir_path: PathBuf| -> anyhow::Result<()> {
-                if !dir_path.is_dir() {
-                    return Ok(());
-                }
+        let view_number = view;
+        let prune_view = ViewNumber::new(view.saturating_sub(self.view_retention));
+
+        let delete_files = |intervals: &[RangeInclusive<ViewNumber>],
```

Comment:
I think it would be a bit more straightforward to make `delete_files` a method on `Inner`. IIUC the only thing it takes from the scope is `prune_view`.